### ISSUE TITLE
Schema override fails for union of [null, A, B]

### ISSUE THREAD

**Issue by RoccoMathijn** [sksamuel/avro4s#521]
I’m trying to override a schema of a encoder but it fails with the error: 

> requirement failed: Schema type for either encoders / decoders must be UNION, received STRING

I broke it down to the bare minimum

```
final case class Foo(bar: Option[Either[String, Int]])

val schemaFor = SchemaFor[Foo]
val encoder = Encoder[Foo].withSchema(schemaFor)

```

If I change bar to just an Either[String,Int] it will work. How I can override with a schema like this?

**Comment by andreas-schroeder** [comment]
It's probably a bug, I will look into it soon.

**Comment by andreas-schroeder** [comment]
@RoccoMathijn it was indeed a bug, fixed with #523 - thanks for reporting!

**Comment by RoccoMathijn** [comment]
Thanks for the fix! Appreciate it
